### PR TITLE
gh-100616: Document 'attr' parameter for window.vline() in curses module

### PR DESCRIPTION
**Pull request by mathieui**
This adds the optional "attr" to the documentation, as it has worked since forever and is also present in the tsts, I do not believe this is significant enough to warrant an issue.

* Issue: gh-100616

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

## Recognized GitHub username

We couldn't find a [bugs.python.org](https://bugs.python.org/) (b.p.o) account corresponding to the following GitHub usernames:

@mathieui

This might be simply due to a missing "GitHub Name" entry in one's b.p.o account settings. This is necessary for legal reasons before we can look at this contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.




You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Review by JelleZijlstra — COMMENTED** [review]
I think documenting a new parameter does need a NEWS entry and bpo link.

Also, the documentation should say what this parameter does.

File: Doc/library/curses.rst [hunk 1]
```diff
@@ -1309,8 +1309,8 @@ the following methods and attributes:
    :meth:`refresh`.
 
 
-.. method:: window.vline(ch, n)
-            window.vline(y, x, ch, n)
+.. method:: window.vline(ch, n[, attr])
+            window.vline(y, x, ch, n[, attr])
 
    Display a vertical line starting at ``(y, x)`` with length *n* consisting of the
    character *ch*.
```
Comment (slateny) [inline Doc/library/curses.rst]:
```suggestion
   character *ch* with attributes *attr*.
```

Following wording from https://github.com/python/cpython/blame/894f2c3c161933bd820ad322b3b678d89bc2377c/Doc/library/curses.rst#L710 (also tested and *attr* does exist, and its values are per the table right under [this section](https://docs.python.org/3/library/curses.html#curses.ncurses_version).

**Comment by JelleZijlstra** [comment]
Thanks @mathieui for the PR and @slateny for finishing it up!

**Pull request merged**